src/module.c

```diff
@@ -25,6 +26,111 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
 
 void ReplyWithSeriesLabels(RedisModuleCtx *ctx, const Series *series);
 
+Label *parseLabelsFromArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, size_t *label_count) {
+    int pos = RMUtil_ArgIndex("LABELS", argv, argc);
+    int first_label_pos = pos + 1;
+    Label *labels = NULL;
+    *label_count = 0;
+    if (pos < 0) {
+        return NULL;
+    }
+    *label_count = (size_t)(max(0, (argc - first_label_pos) / 2 ));
+    if (label_count > 0) {
+        labels = malloc(sizeof(Label) * (*label_count));
+        for (int i=0; i < *label_count; i++) {
+            labels[i].key = RedisModule_CreateStringFromString(NULL, argv[first_label_pos + i*2]);
+            labels[i].value = RedisModule_CreateStringFromString(NULL, argv[first_label_pos + i*2 + 1]);
+        };
+    }
+    return labels;
+}
+
+int parseCreateArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
+                    long long *retentionSecs, long long *maxSamplesPerChunk, size_t *labelsCount, Label **labels) {
+    *retentionSecs = TSGlobalConfig.retentionPolicy;
+    *maxSamplesPerChunk = TSGlobalConfig.maxSamplesPerChunk;
+    *labelsCount = 0;
+    *labels = parseLabelsFromArgs(ctx, argv, argc, labelsCount);;
+
+    if (RMUtil_ArgExists("RETENTION", argv, argc, 1) && RMUtil_ParseArgsAfter("RETENTION", argv, argc, "l", retentionSecs) != REDISMODULE_OK) {
+        return REDISMODULE_ERR;
+    }
+    if (RMUtil_ArgExists("CHUNK_SIZE", argv, argc, 1) && RMUtil_ParseArgsAfter("CHUNK_SIZE", argv, argc, "l", maxSamplesPerChunk) != REDISMODULE_OK) {
+        return REDISMODULE_ERR;
+    }
+    return REDISMODULE_OK;
+}
+
+int _parseAggregationArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, api_timestamp_t *time_delta,
+                         int *agg_type) {
+    RedisModuleString * aggTypeStr = NULL;
```

Comment:
need to add argument number check here. Also, if there is something wrong with the parsing, for example time_delta is not numeric, we return here TSDB_NOTEXISTS and we don't check for this error value later, so the command works